lib/src/publication/remote.dart

```diff
@@ -245,4 +249,21 @@ class RemoteTrackPublication<T extends RemoteTrack>
     }
     participant.room.engine.signalClient.sendUpdateTrackSettings(settings);
   }
+
+  @internal
+  // Update internal var and return true if changed
+  bool updateSubscriptionAllowed(bool allowed) {
+    if (_subscriptionAllowed == allowed) return false;
+    _subscriptionAllowed = allowed;
+    return true;
+  }
+
+  TrackSubscriptionState subscriptionState() {
+    if (!subscribed || !super.subscribed) {
```

Comment:
what is the difference between `subscribed` and `super.subscribed` ?

Reply:
super.subscribed is `track != null`

Reply:
I'm not seeing a getter definition for `RemoteTrackPublication.subscribed`, only a setter. Wouldn't the two behave the same?

Reply:
TrackPublication.subscribed would be the appropriate place to look.

Reply:
Speaking of `subscribed`, @hiroshihorie, the subscribed value should take into account `subscriptionAllowed` and return false if allowed is false.